### PR TITLE
Use numba-cuda>=0.0.13

### PR DESCRIPTION
**Pull request by gmarkall**
## Description
Testing with https://github.com/NVIDIA/numba-cuda on CI.

I am not sure if edits in other repos are required (e.g. I used to have to change an "integration" repo) - any pointers appreciated!

## Checklist
- [X] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [X] New or existing tests cover these changes.
- [ ] The documentation is up to date with these changes.

### CODE REVIEW

**Comment by msarahan** [comment]
This looks right from the recipe edit side. You should only have to manually edit dependencies.yaml, and all the other files should then be generated.

I don't understand the failing builds. I'm having a hard time finding the error that kills the build. Are you seeing this locally?

**Comment by gmarkall** [comment]
Thanks @msarahan - unfortunately I edited the PR before I saw your comment - should I revert that last change? 

I haven't run this locally because I'm not really set up to build / test cudf at the moment (it's not what I usually work on), but will give it a try on Monday.

File: conda/recipes/cudf/meta.yaml
Comment (msarahan) [inline conda/recipes/cudf/meta.yaml:85]: extra space here is tripping it up, I think

```suggestion
    - numba-cuda >=0.0.13
```

**Comment by gmarkall** [comment]
Just noticed the problem, I introduced a space that shouldn't have been there.

File: dependencies.yaml
Comment (gmarkall) [inline dependencies.yaml:568]: IIUC this is assigning to `numba_cuda` - is that the best form for a name to avoid having a dash in a name, or for consistency with the package name should I change this back to:
```suggestion
          - &numba-cuda numba-cuda>=0.0.13
```

with corresponding change below?

Comment (msarahan) [reply dependencies.yaml:568]: I personally don't like using just plain package names as the anchor names, because it makes it harder to figure out  what to look/search for. I prefer things like `numba-cuda-dep`, so that the YAML anchor is distinct from the package name.

That might just be me, though, so I would advise you to follow your best judgement.

Comment (msarahan) [reply dependencies.yaml:568]: One other thing to keep in mind is that a YAML anchor can represent arbitrary chunks of YAML. It is not necessarily just one dependency. A naming formalism might help us with this ambiguity.

Comment (gmarkall) [reply dependencies.yaml:568]: I have no capacity for judgement here (I'm just cargo-cult changing things) so I'll go with your suggestion :-)

```suggestion
          - &numba-cuda-dep numba-cuda>=0.0.13
```

File: dependencies.yaml
Comment (gmarkall) [inline dependencies.yaml:726]:
```suggestion
          - *numba-cuda-dep
```

**Review by msarahan — APPROVED** [review]

**Comment by gmarkall** [comment]
Thanks for looking at this @msarahan - I've just marked as "DO NOT MERGE" for now until I speak to Ben about this later in the Numba weekly sync - I want to first check whether there should be some discussion about the implications of this with other developers - will update here later.

**Comment by gmarkall** [comment]
@msarahan Following the discussion in the cudf meeting last week the consensus seemed to be that this was OK to proceed with, so I've removed the "DO NOT MERGE" label, fixed up the conflicts, and I'm just waiting on CI.

**Comment by vyasr** [comment]
@gmarkall what else do we want to see on this PR before we merge? If we're happy with it, I'd encourage moving forward ASAP so we have some decent runway to test it before the 24.10 release. CC @bdice @brandon-b-miller

**Comment by gmarkall** [comment]
Thanks Vyas - I don't think there's anything else we want to see in it, I just wanted to have made sure to have discussed it with the cuDF team before merging.

**Review by vyasr — APPROVED** [review]

**Comment by vyasr** [comment]
Cool. Going to queue this up for merge then, the dep changes look fine.

**Comment by vyasr** [comment]
Given the delay with the conda-forge recipe and there now being conflicts on this branch, let's push this upgrade to 24.12. That will give us more time to iron out any kinks if anything goes unexpectedly wrong with the switch (particularly if it happens downstream of cudf).

File: dependencies.yaml
Comment (vyasr) [inline dependencies.yaml:723]: Note that I changed this so that we now require numba-cuda even to test the oldest supported dependency. If at some point we choose to support a range of versions then we should remove the alias here and insert a real version.

**Pull request merged**
